cli/impl/src/main/java/io/helidon/build/cli/impl/ArchetypesDataLoader.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.cli.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+import io.helidon.build.common.maven.VersionRange;
+import io.helidon.build.common.xml.SimpleXMLParser;
+import io.helidon.build.common.xml.SimpleXMLParser.Reader;
+
+/**
+ * Loader for {@link ArchetypesData}.
+ * XML reader for archetypes data document.
+ */
+public class ArchetypesDataLoader {
+
+    private static final Map<FileSystem, ArchetypesData> ARCHETYPES_DATA_MAP = new WeakHashMap<>();
+
+    private ArchetypesDataLoader() {
+    }
+
+    /**
+     * Get the data about archetype versions for the given archetype.
+     *
+     * @param archetype archetype
+     * @return data about archetype versions
+     */
+    public static ArchetypesData load(FileSystem archetype) {
+        return ARCHETYPES_DATA_MAP.computeIfAbsent(archetype, ArchetypesDataLoader::archetypesData);
+    }
+
+    private static ArchetypesData archetypesData(FileSystem archetype) {
+        try {
+            Path path = archetype.getPath("versions.xml");
```

Comment:
It's wrong to use a `FileSystem` here.

The archetype `FileSystem` instances are created from the archetype JAR files, however the `versions.xml` file is meant to be included in the cli-data.

I.e. the Metadata class should take care of unpacking `versions.xml` to `~/.helidon/cache`.

The input to this class should be just a `Path`.